dataplane/linux/bpf_ep_mgr.go

```diff
@@ -1344,3 +1392,27 @@ func FindJumpMap(ap *tc.AttachPoint) (mapFD bpf.MapFD, err error) {
 	}
 	return 0, errors.New("failed to find TC program")
 }
+
+func (m *bpfEndpointManager) getInterfaceIP(ifaceName string) (*net.IP, error) {
+	if ip, ok := m.ifaceToIpMap[ifaceName]; ok {
+		return &ip, nil
+	}
+	intf, err := net.InterfaceByName(ifaceName)
+	if err != nil {
+		return nil, err
+	}
+	addrs, err := intf.Addrs()
+	if err != nil {
+		return nil, err
+	}
+	sort.Slice(addrs, func(i, j int) bool {
+		return bytes.Compare(addrs[i].(*net.IPNet).IP, addrs[j].(*net.IPNet).IP) < 0
```

Comment:
The cast makes me wonder if there could be a different type of address in some corner case.  Probably best to filter those out, just in case!